util/regexp-assemble/regexp-942120.data

```diff
@@ -13,7 +13,7 @@ regexp\s+binary
 \bisnull\b
 \brlike\b
 rlike\s+binary
-not\s+between\s+0\s+and
+not\s+between\s+[\d'\"-]+\s+and
```

Comment:
This change is interesting because the original expression guards against hiding `> 0` with `not between 0 and`. The updated expression is a lot more general, maybe too general. It is easy to think of an example in english text that will create an FP: "I'm free this afternoon, but not between 3 and 5 o'clock".

There may still be something here that we can use. First, I would retain the original expression, since it solves one specific evasion. Then, we could add a new expression where we test for two quoted literals around the `and`, much like the test shows. That would still allow for false positives but since this is paranoia level 2, it may be ok.

Reply:
You are amazing, and found a false positive that we missed. So it is to keep the original regularity and add a new regularity `not\s+between\s+['\"][\d-]+['\"]\s+and`. Is my understanding correct?

Reply:
Yes, precisely. Your idea is good, I would suggest to make the expression a little bit more general though. Your expression only matches ISO date format, e.g. `2021-06-12` but the following would pass: `2021-6-12 00:00:00`. Here's my suggestion: `not\s+between\s+(?:(?:['][^']*['])|(?:[\"][^\"]*[\"]))\s+and\s+(?:(?:['][^']*['])|(?:[\"][^\"]*[\"]))`. This matches anything where the two expressions before and after the `and` are quoted and also requires that the same quotes be used (to reduce false positives and also to reduce the likelihood of someone finding a way to evade using the other quote type).

Again, this will still allow for false positives. I'd like to await @franbuehler's opinion on whether that's ok for PL 2.

Reply:
Well spotted, @theseion.
I think PL 2 is ok.

Reply:
So add new rule:
`not\s+between\s+(?:(?:'[^']*')|(?:\"[^\"]*\"))\s+and\s+(?:(?:'[^']*')|(?:\"[^\"]*\"))` 
Revert old rule:
`not\s+between\s+0\s+and`
@franbuehler